mws/apimws/utils.py

```diff
@@ -137,3 +138,26 @@ def domain_confirmation_user(domain_name):
         to=[domain_name.vhost.service.site.email],
         headers={'Return-Path': getattr(settings, 'EMAIL_MWS3_SUPPORT', 'mws-support@uis.cam.ac.uk')}
     ).send()
+
+def next_update(time=None):
+    """
+    Return a time by which we can be reasonably sure the DNS update has run.
+    """
+    from datetime import datetime, timedelta
+    if time is None:
+        time = datetime.now()
+    return time.replace(minute=54, second=0, microsecond=0) + timedelta(hours=1) \
+        if time.minute > 55 \
+        else time.replace(minute=54, second=0, microsecond=0)
+
+
+@shared_task(base=AnsibleTaskWithFailure)
+def configure_domain(domain_name):
+    try:
+        domain = DomainName.objects.get(name=domain_name.name)
+        service = domain.vhost.service
+    except Exception as e:
+        raise Exception(e, domain_name.__class__)
+    from apimws.ansible import launch_ansible
+    launch_ansible(service)
```

Comment:
not sure why this is useful in here?

Reply:
If you mean the try-except, that is a remnant of troubleshooting the redis version bug and could be removed. The Ansible call is required because this will configure the domain the user has set, see comment on mws/sitesmanagement/views/domains.py:99

Reply:
Same comment as above, I think the ACME request shouldn't be triggered until we have some kind of flag (AnsibleConfiguration object or something similar), as an Ansible execution can be trigger by other means. The Ansible code that activates the ACME request should perhaps only be executed once this flag is set up, or maybe only once using an special launch_ansible with an specific tag